providers/azure/resources/cosmosdb.go

```diff
@@ -75,10 +96,92 @@ func (a *mqlAzureSubscriptionCosmosDbService) accounts() ([]interface{}, error)
 					"properties": llx.DictData(properties),
 				})
 			if err != nil {
-				return nil, err
+				return err
 			}
-			res = append(res, mqlCosmosDbAccount)
+			*res = append(*res, mqlCosmosDbAccount)
 		}
 	}
-	return res, nil
+	return nil
+}
+
+func fetchMongoDBClusters(ctx context.Context, runtime *plugin.Runtime, conn *connection.AzureConnection, subId string, res *[]interface{}) error {
+	resClient, err := armresources.NewClient(subId, conn.Token(), &arm.ClientOptions{
+		ClientOptions: conn.ClientOptions(),
+	})
+	if err != nil {
+		return err
+	}
+	filter := "resourceType eq 'Microsoft.DocumentDB/mongoClusters'"
+	pager := resClient.NewListPager(&armresources.ClientListOptions{
+		Filter: &filter,
+	})
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		for _, account := range page.Value {
+			properties, err := convert.JsonToDict(account.Properties)
+			if err != nil {
+				return err
+			}
+
+			mqlResource, err := CreateResource(runtime, "azure.subscription.cosmosDbService.account",
+				map[string]*llx.RawData{
+					"id":         llx.StringDataPtr(account.ID),
+					"name":       llx.StringDataPtr(account.Name),
+					"tags":       llx.MapData(convert.PtrMapStrToInterface(account.Tags), types.String),
+					"location":   llx.StringDataPtr(account.Location),
+					"kind":       llx.StringDataPtr(account.Kind),
+					"type":       llx.StringDataPtr(account.Type),
+					"properties": llx.DictData(properties),
+				})
+			if err != nil {
+				return err
+			}
+			*res = append(*res, mqlResource)
+		}
+	}
+	return nil
+}
+
+func fetchPostgreSQLServerGroups(ctx context.Context, runtime *plugin.Runtime, conn *connection.AzureConnection, subId string, res *[]interface{}) error {
+	resClient, err := armresources.NewClient(subId, conn.Token(), &arm.ClientOptions{
+		ClientOptions: conn.ClientOptions(),
+	})
+	if err != nil {
+		return err
+	}
+	filter := "resourceType eq 'Microsoft.DBforPostgreSQL/serverGroupsv2'"
+	pager := resClient.NewListPager(&armresources.ClientListOptions{
+		Filter: &filter,
+	})
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return err
+		}
+		for _, account := range page.Value {
+			properties, err := convert.JsonToDict(account.Properties)
+			if err != nil {
+				return err
+			}
+
+			mqlResource, err := CreateResource(runtime, "azure.subscription.cosmosDbService.account",
```

Comment:
Won't be need this in the .lr file?

Reply:
No, already exists, we just append for other types!